src/query/auth.ts

```diff
@@ -0,0 +1,30 @@
+import { useMutation } from '@tanstack/react-query';
+import { getMe, postLogin } from 'api/auth';
+import { useNavigate } from 'react-router-dom';
+import useAuthStore from 'store/auth';
+
+import { ZodError } from 'zod';
+
+const useLogin = () => {
+  const navigate = useNavigate();
+  const { setUser } = useAuthStore();
```

Comment:
아래처럼 작성하면 [해당 값의 변경만 재렌더링을 발생시켜 성능을 향샹시킬 수 있다](https://docs.pmnd.rs/zustand/recipes/recipes#fetching-everything)고 하네요.
store에 `user`도 있으니, user의 변경이 setUser가 포함된 해당 훅까지 재렌더링을 발생시킬 수 있는 것이죠.

```suggestion
  const setUser = useAuthStore((state) => state.setUser);
```

Reply:
오 그렇네요 수정하겠습니다ㅎㅎ ff8ab25abcffb21084bc4ccc79cdaedfb4c08265